src/librustc/traits/fulfill.rs

```diff
@@ -256,15 +256,20 @@ impl<'a, 'b, 'tcx> ObligationProcessor for FulfillProcessor<'a, 'b, 'tcx> {
         &mut self,
         pending_obligation: &mut Self::Obligation,
     ) -> ProcessResult<Self::Obligation, Self::Error> {
-        // if we were stalled on some unresolved variables, first check
+        // If we were stalled on some unresolved variables, first check
         // whether any of them have been resolved; if not, don't bother
         // doing more work yet
         if !pending_obligation.stalled_on.is_empty() {
-            if pending_obligation.stalled_on.iter().all(|&ty| {
+            let mut changed = false;
```

Comment:
Maybe add a comment pointing at this PR? This seems like a change that is likely to get reverted by someone cleaning up the code. Something like:

--

// NB: this `for` loop was originally a call to `all`, but rewriting in this lower-level form was found to be a compilation time win.

Reply:
I have done this. Though you should definitely check out #64572, where I have optimized `all()` and a few other iterator methods, with apparently extraordinary effects.

(I also have another pending PR in which I change this loop yet again, specializing it for the common cases of 0 or 1 elements. I am having a good time this week hammering on this part of the code!)